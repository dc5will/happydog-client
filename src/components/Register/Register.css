.RegistrationForm {
    margin: 30px;
}

.registerButton {
    margin: 40px;
    font-size: 24px;
    background-color: rgb(192, 81, 81);
}

.registerButton:hover {
    background-color: rgb(139, 36, 36);
}

.registerButton:active {
    background-color: rgb(138, 32, 32);
    box-shadow: 0 5px #666;
    transform: translateY(2px);
}

label {
    font-size: 20px;
}

h3 {
    margin: 40px;
}