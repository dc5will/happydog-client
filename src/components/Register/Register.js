import React, { Component } from 'react'
import AuthApiService from '../../services/auth-api-service'
import { Section, Button, Required, Input } from '../Utils/Utils';
import Header from '../../components/Header/Header';
import Footer from '../../components/Footer/Footer';
import './Register.css';

class Register extends Component {
  static defaultProps = {
    onRegistrationSuccess: () => {}
  }

  state = { error: null }

  handleSubmit = e => {
    e.preventDefault()
    const { full_name, user_name, password } = e.target
    this.setState({ error: null })
    AuthApiService.postUser({
      user_name: user_name.value,
      password: password.value,
      full_name: full_name.value,
    })
      .then(() => {
        full_name.value = ''
        user_name.value = ''
        password.value = ''
        this.props.history.push('/login')
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }

  render(){
    return(
      <>
      <Header />
      <Section className='Registration'>
        <h2>Welcome to Happydog</h2>
          <h3>Please enter your information to sign up!</h3>
        <form className="RegistrationForm" onSubmit={this.handleSubmit}>
        <div className="full_name">
          <label htmlFor="RegistrationForm_full_name">
            Full name <Required />
          </label>
          <Input
            name="full_name"
            type="text"
            id="RegistrationForm_full_name"
            required>
           </Input>
        </div>

        <div className="user_name">
          <label htmlFor="RegistrationForm_user_name">
            User name <Required />
          </label>
          <Input
            name='user_name'
            type='text'
            id='RegistrationForm_user_name'
            required>
          </Input>
        </div>

        <div className="password">
          <label htmlFor="RegistrationForm_password">
            Password <Required />
          </label>
          <Input
            name='password'
            type='password'
            id='RegistrationForm_password'
            required>
          </Input>
        </div>
        <Button className='registerButton' type='submit'>
            Register
        </Button>
      </form>
      </Section>
      <Footer />
      </>
    )
  }
}

export default Register