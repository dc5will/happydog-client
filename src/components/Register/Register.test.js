import React from "react";
import { shallow } from "enzyme";
import Register from "./Register";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe(`Register component`, () => {
  it("renders without crashing", () => {
    shallow(<Register />);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Register />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});