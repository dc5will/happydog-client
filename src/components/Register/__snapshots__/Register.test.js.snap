// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component renders the UI as expected 1`] = `
Array [
  <nav
    className="Header"
  >
    <h1>
      <a
        href="/my-notes"
        onClick={[Function]}
      >
        Happy dog
      </a>
    </h1>
    <div
      className="Header_not-logged-in"
    >
      <a
        href="/login"
        onClick={[Function]}
      >
        Log in
      </a>
      <a
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </div>
  </nav>,
  <section
    className={
      Array [
        "Section",
        undefined,
        "Registration",
      ]
    }
  >
    <h2>
      Welcome to Happydog
    </h2>
    <h3>
      Please enter your information to sign up!
    </h3>
    <form
      className="RegistrationForm"
      onSubmit={[Function]}
    >
      <div
        className="full_name"
      >
        <label
          htmlFor="RegistrationForm_full_name"
        >
          Full name 
          <span
            className="Required "
          >
            *
          </span>
        </label>
        <input
          className="Input "
          id="RegistrationForm_full_name"
          name="full_name"
          required={true}
          type="text"
        />
      </div>
      <div
        className="user_name"
      >
        <label
          htmlFor="RegistrationForm_user_name"
        >
          User name 
          <span
            className="Required "
          >
            *
          </span>
        </label>
        <input
          className="Input "
          id="RegistrationForm__user_name"
          name="user_name"
          required={true}
          type="text"
        />
      </div>
      <div
        className="password"
      >
        <label
          htmlFor="RegistrationForm_password"
        >
          Password 
          <span
            className="Required "
          >
            *
          </span>
        </label>
        <input
          className="Input "
          id="RegistrationForm__password"
          name="password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="Button "
        type="submit"
      >
        Register
      </button>
    </form>
  </section>,
  <footer
    className="footer"
  >
    <div
      className="footer_brand-container"
    />
  </footer>,
]
`;
